File: TextToSpeech.Web/src/app/dto/voice.ts
export interface Voice {
  name: string;
  providerVoiceId: string;
  language?: Language | null;
  qualityTier?: VoiceQualityTier;
}

interface Language {
  name: string;
  languageCode: string;
}

type VoiceQualityTier = 'Standard' | 'Premium';

File: TextToSpeech.Core/Models/Voice.cs
using static TextToSpeech.Core.Enums;

namespace TextToSpeech.Core.Models;

public sealed class Voice
{
    public required string Name { get; init; }
    public required string ProviderVoiceId { get; init; }
    public Language? Language { get; init; }
    public VoiceQualityTier QualityTier { get; init; }
}

public sealed class Language(string name, string languageCode)
{
    public string Name { get; init; } = name;
    public string LanguageCode { get; init; } = languageCode;
}